base_classes/NXroot.nxdl.xml

```diff
@@ -2,9 +2,9 @@
 <?xml-stylesheet type="text/xsl" href="nxdlformat.xsl"?>
 <!--
 # NeXus - Neutron and X-ray Common Data Format
-# 
-# Copyright (C) 2014-2023 NeXus International Advisory Committee (NIAC)
-# 
+#
+# Copyright (C) 2014-2024 NeXus International Advisory Committee (NIAC)
```

Comment:
How about the starting dates? NXroot has a starting year of 2008 in the currently accepted NIAC [proposal](https://github.com/nexusformat/definitions/blob/main/base_classes/NXroot.nxdl.xml). I don't think we should change this.

See also [this comment](https://github.com/nexusformat/definitions/pull/1303#issuecomment-1807711833) in out original nyaml2nxdl PR on the NIAC repo.

Reply:
There is a discussion inside the NIAC on how to handle these. This PR is not intended to change the starting date or act as a platform for discussing this. It's just meant for updating the files to what `nyaml2nxdl` currently produces. We should then follow up on the general discussion on how to handle and generate copyright dates at a later stage.

Reply:
Ok, makes sense. But we should ensure in the future that `nyaml2nxdl` picks up the correct starting date in agreement with what NIAC is proposing.

What about the first point, i.e., why are only the contributed classes changed? From [this line](https://github.com/FAIRmat-NFDI/nyaml/blob/1437afafd5a10e55121050e764bcf3c883c53f46/nyaml/nyaml2nxdl.py#L53), I would have guesses that all classes are updated to a new datetime? Or is this related to the fact that the yaml files did not change?